In [1]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
df_respiratorio = pd.read_parquet('at_urg_respiratorio_semanal.parquet')
df_calidad_aire =pd.read_csv('Calidad del aire.csv',sep=';')

In [3]:
df_respiratorio.head(4)

EstablecimientoCodigo           EstablecimientoGlosa RegionCodigo  \
0                133160  Hospital Comunitario de Achao           10   
1                200143                 SAPU Aguilucho           13   
2                200744         SUR Huertos Familiares           13   
3                116840               SAR Las Américas            7   

                        RegionGlosa ComunaCodigo  ComunaGlosa  \
0               Región De Los Lagos        10210     Quinchao   
1  Región Metropolitana de Santiago        13123  Providencia   
2  Región Metropolitana de Santiago        13303       Tiltil   
3                  Región Del Maule         7101        Talca   

  ServicioSaludCodigo                       ServicioSaludGlosa  \
0                  33                 Servicio de Salud Chiloé   
1                  12  Servicio de Salud Metropolitano Oriente   
2                   9    Servicio de Salud Metropolitano Norte   
3                  16              Servicio de Salud Del Maule   

                                 TipoEstablecimiento  \
0                                           Hospital   
1   Servicio de Atención Primaria de Urgencia (SAPU)   
2                   Servicio de Urgencia Rural (SUR)   
3  Servicio de Atención Primaria de Urgencia de A...   

  DependenciaAdministrativa  ...  Anio SemanaEstadistica OrdenCausa  \
0         Servicio de Salud  ...  2018                28          8   
1                 Municipal  ...  2021                41          4   
2                 Municipal  ...  2022                35          6   
3                 Municipal  ...  2024                52         11   

                                    Causa NumTotal  NumMenor1Anio  \
0  Crisis obstructiva bronquial (J40-J46)        1              0   
1                      IRA Alta (J00-J06)       17              0   
2                      Neumonía (J12-J18)        1              0   
3      Covid-19, Virus identificado U07.1        0              0   

   Num1a4Anios  Num5a14Anios Num15a64Anios  Num65oMas  
0            0             0             0          1  
1            0             1            14          2  
2            0             1             0          0  
3            0             0             0          0  

[4 rows x 25 columns]

In [4]:
df_respiratorio.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3172850 entries, 0 to 3172849
Data columns (total 25 columns):
 #   Column                     Dtype 
---  ------                     ----- 
 0   EstablecimientoCodigo      object
 1   EstablecimientoGlosa       object
 2   RegionCodigo               object
 3   RegionGlosa                object
 4   ComunaCodigo               object
 5   ComunaGlosa                object
 6   ServicioSaludCodigo        object
 7   ServicioSaludGlosa         object
 8   TipoEstablecimiento        object
 9   DependenciaAdministrativa  object
 10  NivelAtencion              object
 11  TipoUrgencia               object
 12  Latitud                    object
 13  Longitud                   object
 14  NivelComplejidad           object
 15  Anio                       int32 
 16  SemanaEstadistica          int32 
 17  OrdenCausa                 int32 
 18  Causa                      object
 19  NumTotal                   int32 
 20  NumMenor1Anio           

In [5]:
df_respiratorio.isna().sum()

EstablecimientoCodigo        10464
EstablecimientoGlosa         10464
RegionCodigo                 10464
RegionGlosa                  10464
ComunaCodigo                 10464
ComunaGlosa                  10464
ServicioSaludCodigo          10464
ServicioSaludGlosa           10464
TipoEstablecimiento          10464
DependenciaAdministrativa    10464
NivelAtencion                10464
TipoUrgencia                 10464
Latitud                      10464
Longitud                     10464
NivelComplejidad             10464
Anio                             0
SemanaEstadistica                0
OrdenCausa                       0
Causa                            0
NumTotal                         0
NumMenor1Anio                    0
Num1a4Anios                      0
Num5a14Anios                     0
Num15a64Anios                    0
Num65oMas                        0
dtype: int64

In [6]:
df_respiratorio['Anio'] = pd.to_datetime(df_respiratorio['Anio'], format='%Y')

In [7]:
df_respiratorio.describe()

Anio  SemanaEstadistica    OrdenCausa  \
count                        3172850       3.172850e+06  3.172850e+06   
mean   2020-05-05 02:00:15.898388224       2.606733e+01  1.199843e+01   
min              2014-01-01 00:00:00       1.000000e+00  3.000000e+00   
25%              2018-01-01 00:00:00       1.300000e+01  5.000000e+00   
50%              2021-01-01 00:00:00       2.600000e+01  8.000000e+00   
75%              2023-01-01 00:00:00       3.900000e+01  1.100000e+01   
max              2025-01-01 00:00:00       5.300000e+01  3.400000e+01   
std                              NaN       1.502996e+01  1.080485e+01   

           NumTotal  NumMenor1Anio   Num1a4Anios  Num5a14Anios  Num15a64Anios  \
count  3.172850e+06   3.172850e+06  3.172850e+06  3.172850e+06   3.172850e+06   
mean   3.409593e+01   2.152958e+00  6.283911e+00  7.078895e+00   1.556200e+01   
min    0.000000e+00   0.000000e+00  0.000000e+00  0.000000e+00   0.000000e+00   
25%    0.000000e+00   0.000000e+00  0.000000e+00  0.000000e+00   0.000000e+00   
50%    3.000000e+00   0.000000e+00  0.000000e+00  0.000000e+00   1.000000e+00   
75%    2.100000e+01   1.000000e+00  3.000000e+00  3.000000e+00   9.000000e+00   
max    4.460000e+03   2.327000e+03  4.329000e+03  1.016000e+03   1.346000e+03   
std    8.166050e+01   8.145352e+00  1.884002e+01  2.004998e+01   3.890232e+01   

          Num65oMas  
count  3.172850e+06  
mean   3.018161e+00  
min    0.000000e+00  
25%    0.000000e+00  
50%    0.000000e+00  
75%    3.000000e+00  
max    3.080000e+02  
std    6.718043e+00

# Informacion
- El dataset cuenta con 25 columnas
- Tiene 3172850 registros

In [8]:
df_calidad_aire.head()

Estacion  FECHA (YYMMDD)  HORA (HHMM)     MP10 MP2_5   CO   NO  NO2  \
0  Pudahuel          970404            0  114,917   NaN  NaN  NaN  NaN   
1  Pudahuel          970405            0    102,5   NaN  NaN  NaN  NaN   
2  Pudahuel          970406            0  646,666   NaN  NaN  NaN  NaN   
3  Pudahuel          970407            0  799,166   NaN  NaN  NaN  NaN   
4  Pudahuel          970408            0  842,916   NaN  NaN  NaN  NaN   

        O3 Humedad Temperatura Viento_v  
0  354,166     NaN         NaN      NaN  
1  385,833     NaN         NaN      NaN  
2    33,25     NaN         NaN      NaN  
3  400,833     NaN         NaN      NaN  
4  325,833     NaN         NaN      NaN

In [9]:
df_calidad_aire.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 47953 entries, 0 to 47952
Data columns (total 12 columns):
 #   Column          Non-Null Count  Dtype 
---  ------          --------------  ----- 
 0   Estacion        47953 non-null  object
 1   FECHA (YYMMDD)  47953 non-null  int64 
 2   HORA (HHMM)     47953 non-null  int64 
 3   MP10            47227 non-null  object
 4   MP2_5           38031 non-null  object
 5   CO              27641 non-null  object
 6   NO              21308 non-null  object
 7   NO2             18368 non-null  object
 8   O3              27937 non-null  object
 9   Humedad         19656 non-null  object
 10  Temperatura     20154 non-null  object
 11  Viento_v        20262 non-null  object
dtypes: int64(2), object(10)
memory usage: 4.4+ MB


In [10]:
df_calidad_aire['FECHA (YYMMDD)'] =df_calidad_aire['FECHA (YYMMDD)'].astype(str)
df_calidad_aire['FECHA (YYMMDD)'] = df_calidad_aire['FECHA (YYMMDD)'].str.zfill(6)
df_calidad_aire['FECHA (YYMMDD)'] = pd.to_datetime(df_calidad_aire['FECHA (YYMMDD)'], format='%y%m%d')
df_calidad_aire.head()

Estacion FECHA (YYMMDD)  HORA (HHMM)     MP10 MP2_5   CO   NO  NO2  \
0  Pudahuel     1997-04-04            0  114,917   NaN  NaN  NaN  NaN   
1  Pudahuel     1997-04-05            0    102,5   NaN  NaN  NaN  NaN   
2  Pudahuel     1997-04-06            0  646,666   NaN  NaN  NaN  NaN   
3  Pudahuel     1997-04-07            0  799,166   NaN  NaN  NaN  NaN   
4  Pudahuel     1997-04-08            0  842,916   NaN  NaN  NaN  NaN   

        O3 Humedad Temperatura Viento_v  
0  354,166     NaN         NaN      NaN  
1  385,833     NaN         NaN      NaN  
2    33,25     NaN         NaN      NaN  
3  400,833     NaN         NaN      NaN  
4  325,833     NaN         NaN      NaN

In [11]:
df_calidad_aire =df_calidad_aire.rename(columns={'FECHA (YYMMDD)': 'fecha'})

In [12]:
df_calidad_aire.describe()

fecha  HORA (HHMM)
count                          47953      47953.0
mean   2010-06-04 09:02:59.500761088          0.0
min              1997-04-02 00:00:00          0.0
25%              2003-11-11 00:00:00          0.0
50%              2010-06-04 00:00:00          0.0
75%              2016-12-27 00:00:00          0.0
max              2023-07-27 00:00:00          0.0
std                              NaN          0.0

In [13]:
df_calidad_aire['MP10']=df_calidad_aire['MP10'].str.replace(',', '.', regex=False).astype(float)
df_calidad_aire['MP2_5']=df_calidad_aire['MP2_5'].str.replace(',', '.', regex=False).astype(float)
df_calidad_aire['CO']=df_calidad_aire['CO'].str.replace(',', '.', regex=False).astype(float)
df_calidad_aire['NO']=df_calidad_aire['NO'].str.replace(',', '.', regex=False).astype(float)
df_calidad_aire['NO2']=df_calidad_aire['NO2'].str.replace(',', '.', regex=False).astype(float)
df_calidad_aire['O3']=df_calidad_aire['O3'].str.replace(',', '.', regex=False).astype(float)
df_calidad_aire['Humedad']=df_calidad_aire['Humedad'].str.replace(',', '.', regex=False).astype(float)
df_calidad_aire['Temperatura']=df_calidad_aire['Temperatura'].str.replace(',', '.', regex=False).astype(float)
df_calidad_aire['Viento_v']=df_calidad_aire['Viento_v'].str.replace(',', '.', regex=False).astype(float)



df_calidad_aire.head()

Estacion      fecha  HORA (HHMM)     MP10  MP2_5  CO  NO  NO2       O3  \
0  Pudahuel 1997-04-04            0  114.917    NaN NaN NaN  NaN  354.166   
1  Pudahuel 1997-04-05            0  102.500    NaN NaN NaN  NaN  385.833   
2  Pudahuel 1997-04-06            0  646.666    NaN NaN NaN  NaN   33.250   
3  Pudahuel 1997-04-07            0  799.166    NaN NaN NaN  NaN  400.833   
4  Pudahuel 1997-04-08            0  842.916    NaN NaN NaN  NaN  325.833   

   Humedad  Temperatura  Viento_v  
0      NaN          NaN       NaN  
1      NaN          NaN       NaN  
2      NaN          NaN       NaN  
3      NaN          NaN       NaN  
4      NaN          NaN       NaN

In [14]:
df_calidad_aire.fillna(0, inplace=True)
df_calidad_aire.head(5)

Estacion      fecha  HORA (HHMM)     MP10  MP2_5   CO   NO  NO2       O3  \
0  Pudahuel 1997-04-04            0  114.917    0.0  0.0  0.0  0.0  354.166   
1  Pudahuel 1997-04-05            0  102.500    0.0  0.0  0.0  0.0  385.833   
2  Pudahuel 1997-04-06            0  646.666    0.0  0.0  0.0  0.0   33.250   
3  Pudahuel 1997-04-07            0  799.166    0.0  0.0  0.0  0.0  400.833   
4  Pudahuel 1997-04-08            0  842.916    0.0  0.0  0.0  0.0  325.833   

   Humedad  Temperatura  Viento_v  
0      0.0          0.0       0.0  
1      0.0          0.0       0.0  
2      0.0          0.0       0.0  
3      0.0          0.0       0.0  
4      0.0          0.0       0.0

# Preguntas

In [15]:
filtered_causes_df = df_respiratorio[~df_respiratorio['Causa'].isin(['TOTAL CAUSAS SISTEMA RESPIRATORIO', 'HOSPITALIZACIONES POR CAUSAS SISTEMA RESPIRATORIO', 'HOSPITALIZACIONES COVID-19, VIRUS IDENTIFICADO U07.1', 'HOSPITALIZACIONES COVID-19, VIRUS NO IDENTIFICADO U07.2'])]
top_5_enfermedade = filtered_causes_df.groupby("Causa")["NumTotal"].sum().sort_values(ascending=False).head(5)
top_5_enfermedade

Causa
IRA Alta (J00-J06)                                      33464799
Bronquitis/bronquiolitis aguda (J20-J21)                 7261966
Covid-19, Virus no identificado U07.2                    4909392
Otra causa respiratoria (J22, J30-J39, J47, J60-J98)     4811223
Crisis obstructiva bronquial (J40-J46)                   2421866
Name: NumTotal, dtype: int32

In [16]:

personas_65_mas = filtered_causes_df.groupby('EstablecimientoGlosa')['Num65oMas'].sum().sort_values(ascending=False).head(10)
personas_65_mas

EstablecimientoGlosa
Hospital Padre Bernabé de Lucerna (Panguipulli) (D)         37278
SAR Renca                                                   30653
Complejo Asistencial Dr. Víctor Ríos Ruiz (Los Ángeles)     29704
SAR Conchalí                                                28969
SAPU La Granja                                              28957
Complejo Hospitalario San José (Santiago, Independencia)    27363
Hospital San José (Coronel)                                 27173
Hospital Barros Luco Trudeau (Santiago, San Miguel)         26941
Hospital Las Higueras (Talcahuano)                          26447
Hospital de San Carlos                                      25812
Name: Num65oMas, dtype: int32

In [17]:
estacion = df_calidad_aire.groupby('Estacion')[['MP10'	,'MP2_5',	'CO'	,'NO',	'NO2',	'O3']].mean()
estacion

MP10       MP2_5         CO          NO         NO2  \
Estacion                                                                
El Bosque   315.999101   79.893596   0.000000    0.000000    0.000000   
La Florida  313.502861  155.894948   0.000000    0.000000    0.000000   
Las Condes  299.353737  128.705342  10.915948  288.843488  109.682746   
Pudahuel    293.228426  155.276266  35.622926  307.314519  125.537911   
Santiago    302.046441  159.526216  38.401089  189.284849   88.290581   

                    O3  
Estacion                
El Bosque     0.000000  
La Florida    0.000000  
Las Condes  189.455122  
Pudahuel    189.869547  
Santiago    172.901034

In [18]:
df_calidad_aire['fecha'] = pd.to_datetime(df_calidad_aire['fecha'])
df_calidad_aire['anio'] = df_calidad_aire['fecha'].dt.year


In [19]:
df_calidad_aire.head()

Estacion      fecha  HORA (HHMM)     MP10  MP2_5   CO   NO  NO2       O3  \
0  Pudahuel 1997-04-04            0  114.917    0.0  0.0  0.0  0.0  354.166   
1  Pudahuel 1997-04-05            0  102.500    0.0  0.0  0.0  0.0  385.833   
2  Pudahuel 1997-04-06            0  646.666    0.0  0.0  0.0  0.0   33.250   
3  Pudahuel 1997-04-07            0  799.166    0.0  0.0  0.0  0.0  400.833   
4  Pudahuel 1997-04-08            0  842.916    0.0  0.0  0.0  0.0  325.833   

   Humedad  Temperatura  Viento_v  anio  
0      0.0          0.0       0.0  1997  
1      0.0          0.0       0.0  1997  
2      0.0          0.0       0.0  1997  
3      0.0          0.0       0.0  1997  
4      0.0          0.0       0.0  1997

In [20]:
df_respiratorio['anio'] = df_respiratorio['Anio'].dt.year

In [21]:
df_merge = pd.merge(df_respiratorio,df_calidad_aire,left_on=['ComunaGlosa','anio'],right_on=['Estacion','anio'],how='right')
df_merge.head()

EstablecimientoCodigo EstablecimientoGlosa RegionCodigo RegionGlosa  \
0                   NaN                  NaN          NaN         NaN   
1                   NaN                  NaN          NaN         NaN   
2                   NaN                  NaN          NaN         NaN   
3                   NaN                  NaN          NaN         NaN   
4                   NaN                  NaN          NaN         NaN   

  ComunaCodigo ComunaGlosa ServicioSaludCodigo ServicioSaludGlosa  \
0          NaN         NaN                 NaN                NaN   
1          NaN         NaN                 NaN                NaN   
2          NaN         NaN                 NaN                NaN   
3          NaN         NaN                 NaN                NaN   
4          NaN         NaN                 NaN                NaN   

  TipoEstablecimiento DependenciaAdministrativa  ... HORA (HHMM)     MP10  \
0                 NaN                       NaN  ...           0  114.917   
1                 NaN                       NaN  ...           0  102.500   
2                 NaN                       NaN  ...           0  646.666   
3                 NaN                       NaN  ...           0  799.166   
4                 NaN                       NaN  ...           0  842.916   

  MP2_5   CO   NO  NO2       O3  Humedad Temperatura  Viento_v  
0   0.0  0.0  0.0  0.0  354.166      0.0         0.0       0.0  
1   0.0  0.0  0.0  0.0  385.833      0.0         0.0       0.0  
2   0.0  0.0  0.0  0.0   33.250      0.0         0.0       0.0  
3   0.0  0.0  0.0  0.0  400.833      0.0         0.0       0.0  
4   0.0  0.0  0.0  0.0  325.833      0.0         0.0       0.0  

[5 rows x 38 columns]

In [26]:
df_merge.sample(10)

EstablecimientoCodigo           EstablecimientoGlosa RegionCodigo  \
31585252                114816  SAPU Dr. Fernando Maffioletti           13   
6667053                 201015              SAR Violeta Parra           13   
34311315                114831                SAPU La Florida           13   
347013                  200139         SAPU Pudahuel Poniente           13   
27906686                114805             SAPU Los Quillayes           13   
7200844                 112806            SAPU Aníbal Ariztía           13   
14488652                111800           SAPU Consultorio Nº1           13   
6822285                 200139         SAPU Pudahuel Poniente           13   
32062428                114804           SAPU Villa O'Higgins           13   
13136165                111800           SAPU Consultorio Nº1           13   

                               RegionGlosa ComunaCodigo ComunaGlosa  \
31585252  Región Metropolitana de Santiago        13110  La Florida   
6667053   Región Metropolitana de Santiago        13124    Pudahuel   
34311315  Región Metropolitana de Santiago        13110  La Florida   
347013    Región Metropolitana de Santiago        13124    Pudahuel   
27906686  Región Metropolitana de Santiago        13110  La Florida   
7200844   Región Metropolitana de Santiago        13114  Las Condes   
14488652  Región Metropolitana de Santiago        13101    Santiago   
6822285   Región Metropolitana de Santiago        13124    Pudahuel   
32062428  Región Metropolitana de Santiago        13110  La Florida   
13136165  Región Metropolitana de Santiago        13101    Santiago   

         ServicioSaludCodigo                           ServicioSaludGlosa  \
31585252                  14  Servicio de Salud Metropolitano Sur Oriente   
6667053                   10    Servicio de Salud Metropolitano Occidente   
34311315                  14  Servicio de Salud Metropolitano Sur Oriente   
347013                    10    Servicio de Salud Metropolitano Occidente   
27906686                  14  Servicio de Salud Metropolitano Sur Oriente   
7200844                   12      Servicio de Salud Metropolitano Oriente   
14488652                  11      Servicio de Salud Metropolitano Central   
6822285                   10    Servicio de Salud Metropolitano Occidente   
32062428                  14  Servicio de Salud Metropolitano Sur Oriente   
13136165                  11      Servicio de Salud Metropolitano Central   

                                        TipoEstablecimiento  \
31585252   Servicio de Atención Primaria de Urgencia (SAPU)   
6667053   Servicio de Atención Primaria de Urgencia de A...   
34311315   Servicio de Atención Primaria de Urgencia (SAPU)   
347013     Servicio de Atención Primaria de Urgencia (SAPU)   
27906686   Servicio de Atención Primaria de Urgencia (SAPU)   
7200844    Servicio de Atención Primaria de Urgencia (SAPU)   
14488652   Servicio de Atención Primaria de Urgencia (SAPU)   
6822285    Servicio de Atención Primaria de Urgencia (SAPU)   
32062428   Servicio de Atención Primaria de Urgencia (SAPU)   
13136165   Servicio de Atención Primaria de Urgencia (SAPU)   

         DependenciaAdministrativa  ...     MP10    MP2_5        CO       NO  \
31585252                 Municipal  ...   38.000   10.000  0.000000    0.000   
6667053                  Municipal  ...   45.000   13.000  0.780000  132.035   
34311315                 Municipal  ...  114.000   65.000  0.000000    0.000   
347013                   Municipal  ...  509.583  407.083  0.425167  221.577   
27906686                 Municipal  ...   65.000  188.334  0.000000    0.000   
7200844                  Municipal  ...  493.125   16.125  0.256566  521.753   
14488652         Servicio de Salud  ...    0.000    0.000  0.000000    0.000   
6822285                  Municipal  ...  146.000   71.000  1.420000  761.027   
32062428                 Municipal  ...   69.000   16.000  0.000000    0.000   
13136165         Servicio 

In [23]:
df_merge['num_semana'] = df_merge['fecha'].dt.isocalendar().week

In [24]:
prueba = df_merge[df_merge['Causa']=='IRA Alta (J00-J06)']['MP2_5'].mean()
prueba

98.5878856855587

In [ ]:
filtro_df = df_merge[df_merge['']]